### ISSUE TITLE
Meteor Packages - Npm.depends() fails for some modules.

### ISSUE THREAD

**Issue by theodorDiaconu** [meteor/meteor#8028]
I have a package. That I want to run tests for. And I need some npm modules for it. I require them like this:

```
Npm.depends({
    'sift': '3.2.6',
    'redis': '2.6.3',
    'dot-object': '1.5.4'
});
```

It fails for redis. For others it works. Error log:
http://pastebin.com/7qeJzXaj

If I install it in my root:
```
meteor npm install redis
```
And then I omit "depending" it in package.js. It works.

Reproduce:
https://github.com/cult-of-coders/redis-oplog/tree/repro-bug

Update:
This only happens when testing:
```
meteor test-packages --driver-package practicalmeteor:mocha packages/redis-oplog/
```

Update 2:
Maybe it's because while testing, linking is done differently ? Same thing happens with dispatchjs:phantom.

**Comment by abernix** [comment]
What if you try using an NPM peer-dependency approach instead, explained by @hwillson's comment [in this thread](https://github.com/meteor/meteor/issues/3592#issuecomment-259027206):

> The current Meteor package peer npm depenency recommendation is to avoid using `Npm.depends` (due to its fairly rigid nature) and instead leverage a solution that is more inline with npm's [peer dependencies](https://nodejs.org/en/blog/npm/peer-dependencies/) approach. The [Meteor Guide](https://guide.meteor.com/writing-atmosphere-packages.html#peer-npm-dependencies) currently recommends using something like the [tmeasday:check-npm-versions](https://atmospherejs.com/tmeasday/check-npm-versions) package to help ensure package users have npm installed any necessary dependencies.

**Comment by theodorDiaconu** [comment]
@abernix I want to do so. I actually added it like that in the beginning, however, I want continuous integration for testing and it doesn't find the module. I'll play more with travis, maybe there is an easy work-around.

**Comment by theodorDiaconu** [comment]
My work around was to install 1.4.1.3 for testing. #7999 was the issue which will be fixed in 1.4.2.1.
